core/src/main/java/org/htsjdk/core/api/PathURI.java

```diff
@@ -0,0 +1,86 @@
+package org.htsjdk.core.api;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.Path;
+
+/**
+ * Interface representing htsjdk-next input/output resources as URIs.
+ */
+public interface PathURI {
+
+    /**
+     * @return true if this URI has a scheme that has an installed NIO file system provider. This does not
+     * guarantee the URI can be converted into a {@code java.nio.file.Path}, since the URI can be syntactically
+     * valid, and specify a valid file system provider, but still fail to be semantically meaningful.
+     */
+    boolean isNIO();
+
+    /**
+     * Return true if this {code PathURI} can be resolved to an NIO Path. If true, {@code #toPath()} can be
+     * safely called.
+     *
+     * There are cases where a valid URI with a valid scheme backed by an installed NIO File System
+     * still can't be turned into a {@code java.nio.file.Path}, i.e., the following specifies an invalid
+     * authority "namenode":
+     *
+     *  hdfs://namenode/to/file
+     *
+     * The current implementation returns false for these cases (toPath will fail, getInvalidPathReason
+     * returns the reason code).
+     */
+    boolean isPath();
+
+    /**
+     * Get a {@code java.net.URI} object for this {@code PathURI}. Will not be null.
+     * @return The {@code URI} object for this PathURI.
+     */
+    URI getURI();
+
+    /**
+     * Returns the string from which this {@code PathURI} was originally created. This string may differ from
+     * the normalized string returned from a Path that has been object resolved from this PathURI.
+     * @return string from which this URI as originally created. Will not be null, and will always
+     * include a URI scheme.
+     */
+    String getURIString();
+
+    /**
+     * Return the raw input string provided to the constructor.
+     */
+    String getRawInputString();
+
+    /**
+     * Resolve this PathURI to an NIO Path. Can be safely called only if {@code #isPath()} returns true.
+     */
+    Path toPath();
```

Comment:
As explained before, I think that it is better to remove that many `isNio`/`isPath` and throw custom exceptions that can be catch by client code.